📖 doc: extend quickstart to support ClusterClass (#5839)
kubernetes-sigs/cluster-api: docs/book/src/user/quick-start.md

automatically adds to the list the `cluster-api` core provider, and if unspecified, it also adds the `kubeadm` bootstrap
and `kubeadm` control-plane providers.

#### Enabling Feature Gates

Feature gates can be enabled by exporting environment variables before executing `clusterctl init`. 

> For example, the `ClusterTopology` feature, which is required to managed Clusters with ClusterClass,
> can be enabled via:

Reviewer: nit
```suggestion
For example, the `ClusterTopology` feature, which is required to enable support for managed topologies and ClusterClass,
can be enabled via:
```

Author: Done

#### Initialization for common providers

<aside class="note">

<h1>ClusterClass</h1>

If you want to deploy a Cluster with ClusterClass, the `ClusterTopology` feature has to be enabled.
```bash
export CLUSTER_TOPOLOGY=true

> ```

Reviewer: Let's make this more generic by having a sub paragraph named "enabling feature gated features" that links to the feature gate doc, but keeping ClusterClass as example

Author: Done, ptal
